db/migrate/20150910153517_add_full_name_to_custom_attributes.rb

```diff
@@ -0,0 +1,5 @@
+class AddFullNameToCustomAttributes < ActiveRecord::Migration
+  def change
+    add_column :custom_attributes, :full_name, :text
```

Comment:
source is for when you have multiple sets of custom attributes.  For example we have "VC" which are the ones that come from vCenter, and we have "MIQ" or "EMS" which are the ones used for our own application.

I don't quite understand why you need a unique key, but I'm thinking it *might* need the source as well, unless the caller handles the source part of the query.